### ISSUE TITLE
RedirectResponse issue

### ISSUE THREAD

**Issue by lashus** [korotovsky/SingleSignOnServiceProviderBundle#11]
Hello,

I've been trying to implement your bundle but I think there is something wrong with the AuthenticationEntryPoint. After some debugging I found that it tries to create external redirect uri using symfony's HttpUtils::createRedirectResponse() method. The problem is - this method doesnt allow creating external redirects.

Please take a look at this:

**Comment by korotovsky** [comment]
Hi, what do you mean with "external"? AFAIK it worked well at least 2 years ago :) Maybe Symfony team has changed something?

**Comment by lashus** [comment]
Yeah that's what I'm worried about. By external I mean generating a route to "idp.example.com" from "sp.example.com".

**Comment by korotovsky** [comment]
OK, I see. We either could properly configure SSO bundle and set correct `$domainRegexp` and a set of IdP and all SPs domains, or just fix it somehow else...

**Comment by lashus** [comment]
Well perhaps we can manage to do it with oneliner like this:

```php
return new RedirectResponse($this->httpUtils->generateUri($request, $redirectUri), 302);
```

instead of 

```php
return $this->httpUtils->createRedirectResponse($request, $redirectUri);
```

**Comment by korotovsky** [comment]
Concerning security I'd rather set correct `domainRegex`... in the end it's SSO bundle :) I'd limit the amount  of domain where you could be redirected in theory

**Comment by lashus** [comment]
Well, that's certainly a point and probably a reason why Symfony entered this. Therefore there's nothing to include in a bundle except for a change in docs perhaps. The httpUtils can have domainRegex set up but it cannot be changed globally because potentially you can break the site.

The only reasonable way to do that would be factoring a new HttpUtils instance with proper domainRegex just to inject it onto SSO bundle. Although I think it would be easier to set our own Regex to simply match the value in config?

**Comment by korotovsky** [comment]
Yes, factoring is a great idea. Maybe we could extend DI Configuration and ask developers to provide a so called white list of the domains and then build a regex out of it

**Comment by lashus** [comment]
Hmmm, isn't it a potential security breach if they add a method like 'setDomainRegex" or simply 'setHttpUtils' to authenticator class?

I think the best approach would be to abandon "redirectresponse" method from symfony HttpUtils class and create it internally in UriSigner or even in a new class that has HttpUtils injected (and use generateUri from there).

Something like this:

```php


class SsoHttpUtils {

    /**
     * Native symfony http utils. This class is a helper and a bypass checker method for valid RedirectResponse generation.
     *
     * @var \Symfony\Component\Http\HttpUtils $httpUtils
     */
    protected $httpUtils;

    /**
     * @var array
     */
    protected $whitelistDomains;

    public function __construct(HttpUtils $httpUtils, $whitelistDomains = [])
    {
        $this->httpUtils = $httpUtils;
        $this->whitelistDomains = $whitelistDomains;
    }

    public function createRedirectResponse(Request $request, $path, $status = 302)
    {

        if(!$this->isValidDomain($path)) {
            throw new \InvalidArgumentException('Domain is not whitelisted');
        }

        return new RedirectResponse($this->generateUri($request, $path), $status);
    }

    protected function isValidDomain($url) {
        // validation of domains here
        foreach($this->whitelistDomains as $domain) {

            $domainRegexp = '#%s#i';
            if (null !== $domain && preg_match('#^https?://[^/]++#i', $path, $host) && !preg_match(sprintf($domainRegexp, preg_quote($request->getHttpHost())), $host[0])) {
                $path = '/';
            }
        }

        return false;
    }

}
```

**Comment by korotovsky** [comment]
Ok, let's try to go this way
